src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionMappingService.cs

```diff
@@ -345,13 +346,13 @@ private LocationMappingPlan CreateLocationMappings(
                     return new LocationLevelMappings
                     {
                         Mappings = level
-                            .Options
-                            .Select(option => option.ToRow())
+                            .OptionLinks
                             .ToDictionary(
-                                keySelector: MappingKeyFunctions.LocationOptionMetaRowKeyGenerator,
-                                elementSelector: option => new LocationOptionMapping
+                                keySelector: MappingKeyFunctions.LocationOptionMetaLinkKeyGenerator,
+                                elementSelector: link => new LocationOptionMapping
                                 {
-                                    Source = CreateLocationOptionFromMetaRow(option)
+                                    PublicId = link.PublicId,
```

Comment:
Just wondering if this could have been avoided if we'd made `PublicId` required? 

Seems like there's a few fields that haven't been marked as required on the model, so not sure if there's any particular reason for this. If not, would probably be a good idea to make as many fields required as possible!

Similar thing applies for the filter mapping model as well.

Reply:
Ah yeah, good point. I think you are right and I tried making this change yesterday but it upset a few things and ran out of time getting it sorted. In the mean time though I pushed another change related to the refactoring of MappingKeyFunctions that appeared to have been made before but lost.